packages/pds/src/db/index.ts

```diff
@@ -55,20 +58,61 @@ export class Database {
       dialect: new PostgresDialect({ pool }),
     })
 
-    return new Database(db, 'pg', schema)
+    return new Database(db, { dialect: 'pg', pool, schema })
   }
 
   static memory(): Database {
     return Database.sqlite(':memory:')
   }
 
+  notify(channel: string, msg?: string) {
+    if (this.facets.dialect === 'pg') {
+      this.facets.pool.query(`NOTIFY ${channel}, '${msg}'`)
+    } else {
+      const cbs = this.listeners[channel]
+      if (cbs && cbs.length > 0) {
+        for (const cb of cbs) {
+          cb(msg)
+        }
+      }
+    }
+  }
+
+  async listenFor(channel: string, cb: ListenerCB) {
+    if (this.facets.dialect === 'pg') {
+      if (!this.notifyClient) {
+        this.notifyClient = await this.facets.pool.connect()
```

Comment:
We might want some reconnection logic for the notify client to really ensure reliability.

Reply:
stuck some reconnection logic in there that is fairly basic, but i had a rough time trying to test it 🙃